src/Microsoft.AspNet.Mvc.DataAnnotations/DataTypeClientModelValidatorProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    public class DataTypeClientModelValidatorProvider : IClientModelValidatorProvider
+    {
+        private static readonly HashSet<Type> _numericTypes = new HashSet<Type>(new Type[]
+        {
+            typeof(byte), typeof(sbyte),
+            typeof(short), typeof(ushort),
+            typeof(int), typeof(uint),
+            typeof(long), typeof(ulong),
+            typeof(float), typeof(double), typeof(decimal)
```

Comment:
Understand covering `float`, `double` and `decimal` here. But are the remaining `Type`s (all of which should result in a `type="number"` attribute on the element) necessary? I haven't seen issues validating integers and am wondering what I'm missing.